bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -180,9 +180,11 @@ public Connection(@Nullable Connection oldConnection, Gson gson) {
         } else {
             // generate device id
             StringBuilder deviceIdBuilder = new StringBuilder();
-            for (int i = 0; i < 64; i++) {
-                deviceIdBuilder.append(rand.nextInt(9));
-            }
+            byte[] bytes = new byte[16];
+            rand.nextBytes(bytes);
+            String hexStr = HexUtils.bytesToHex(bytes).toUpperCase();
+            hexStr = HexUtils.bytesToHex(hexStr.getBytes());
+            deviceIdBuilder.append(hexStr);
             deviceIdBuilder.append("23413249564c5635564d32573831");
             this.deviceId = deviceIdBuilder.toString();
```

Comment:
```
byte[] bytes = new byte[16];
rand.nextBytes(bytes);
String hexStr = HexUtils.bytesToHex(bytes).toUpperCase();
this.deviceId = HexUtils.bytesToHex(hexStr) + "23413249564c5635564d32573831";
```
```

Reply:
Oh yes, thats correct. I will change it.

Reply:
Maybe i am wrong but every plus between strings use a StringBuilder!?!